Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+        if (Settings.getValue(SETTING_KEY) === entityID) {
+            standUp();
+        }
+
+        if (overlayIntervalTransparency !== null) {
+            Script.clearInterval(overlayIntervalTransparency);
+            overlayIntervalTransparency = null;
+        }
+
+        if (overlayCheckForHover !== null) {
+            Script.clearInterval(overlayCheckForHover);
+            overlayCheckForHover = null;
+        }
+
+        if (overlaySittable) overlays.sittable.remove();
+        if (overlayPreSit) overlays.preSit.remove();
+        if (overlayStandUp) overlays.standUp.remove();
+
+    };
+
+    function sitAndPinAvatar() {
+
+        Settings.setValue(SETTING_KEY, entityID);
+        sitDownSettlePeriod = Date.now() + IK_SETTLE_TIME;
+
+        print(4);
+
+        // MyAvatar.resetSensorsAndBody();
+
+        print(5);
+        MyAvatar.characterControllerEnabled = false;
+        MyAvatar.hmdLeanRecenterEnabled = false;
+
+        var roles = utils.rolesToOverride();
+
+        for (i in roles) { // restore roles to prevent overlap
+            MyAvatar.restoreRoleAnimation(roles[i]);
+            MyAvatar.overrideRoleAnimation(roles[i], ANIMATION_URL, ANIMATION_FPS, true,
+                ANIMATION_FIRST_FRAME, ANIMATION_LAST_FRAME);
+        }
+        for (var i in OVERRIDDEN_DRIVE_KEYS) {
+            MyAvatar.disableDriveKey(OVERRIDDEN_DRIVE_KEYS[i]);
+        }
+
+        MyAvatar.centerBody();
+        var index = MyAvatar.getJointIndex("Hips");
+
+        // var hipLocalRotation = MyAvatar.getJointRotation(index);
+        // var hipWorldRotation = Quat.multiply(MyAvatar.orientation, hipLocalRotation);
+        // var yawHipWorldRotation = Quat.cancelOutRollAndPitch(hipWorldRotation);
+
+        Script.setTimeout(function () {
+
+            print(6, JSON.stringify(chairProperties.rotation));
+
+            if ((HMD.active || testing) && showsPresitOverlay) {
+                Script.update.disconnect(overlays.preSit.update);
+                overlays.preSit.remove();
+            }
+
+            sittingDown = true;
+
+            MyAvatar.pinJoint(index, seatCenterPosition, chairProperties.rotation);
+            canStand = true;
+
+            Script.update.connect(update);
+            MyAvatar.scaleChanged.connect(standUp);
+            location.hostChanged.connect(standUp);
+
+        }, SIT_DELAY);
+
+        animStateHandlerID = MyAvatar.addAnimationStateHandler(function (avatarProperties) {
+            return { headType: 0 };
+        }, ["headType"]);
+
+    }
+
+    function sitDown() {
+        print("SIT DOWN");
+
+        // if (utils.checkSeatForAvatar()) {
+        //     print("Someone is already sitting in that chair.");
+        //     return;
+        // }
+
+        if (overlaySittable !== null) {
+            overlays.sittable.remove();
+        }
+
+        utils.setHeadToHipsDistance();
+
+        lockChairOnStandUp = Entities.getEntityProperties(entityID, 'locked').locked;
```

Comment:
Why do we need to lock/unlock the chair, it doesn't look like we are modifying it at all.